From 45a5e8008ee2edf7870685c71fe947cc37ce2853 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Wed, 4 Dec 2024 09:22:00 -0700
Subject: [PATCH 04/14] FROMGIT: fixup! netfs: Change the read result collector
 to only use one work item

This is basically:

  $ git diff vfs/vfs.git#vfs-6.14.netfs..dhowells/linux-fs#netfs-writeback

Link: https://lore.kernel.org/589335.1733244955@warthog.procyon.org.uk/
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 fs/netfs/read_retry.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/fs/netfs/read_retry.c b/fs/netfs/read_retry.c
index 8ca0558570c1..f0602ca78fcb 100644
--- a/fs/netfs/read_retry.c
+++ b/fs/netfs/read_retry.c
@@ -42,8 +42,6 @@ static void netfs_retry_read_subrequests(struct netfs_io_request *rreq)
 	 */
 	if (!rreq->netfs_ops->prepare_read &&
 	    !test_bit(NETFS_RREQ_COPY_TO_CACHE, &rreq->flags)) {
-		struct netfs_io_subrequest *subreq;
-
 		list_for_each_entry(subreq, &stream->subrequests, rreq_link) {
 			if (test_bit(NETFS_SREQ_FAILED, &subreq->flags))
 				break;
@@ -72,7 +70,7 @@ static void netfs_retry_read_subrequests(struct netfs_io_request *rreq)
 	next = stream->subrequests.next;
 
 	do {
-		struct netfs_io_subrequest *subreq = NULL, *from, *to, *tmp;
+		struct netfs_io_subrequest *from, *to, *tmp;
 		struct iov_iter source;
 		unsigned long long start, len;
 		size_t part;
-- 
2.47.1

